### PR TITLE
Add a gRPC client in C++ for inception serving.

### PR DESCRIPTION
**Pull request by isiosia**
#281 

This is a gRPC client which written in C++ and can run with the inception serving demo.

Build this demo with the following command:

bazel build tensorflow_serving/example:inception_client_cc

And run:

bazel-bin/tensorflow_serving/example/inception_client_cc --image=/file/to/my_cat.jpg

### CODE REVIEW

File: tensorflow_serving/example/inception_client.cc [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+using grpc::ClientReaderWriter;
+using grpc::ClientWriter;
+using grpc::Status;
+
+
+using tensorflow::serving::PredictRequest;
+using tensorflow::serving::PredictResponse;
+using tensorflow::serving::PredictionService;
+
+const std::string SERVING_HOST = "localhost:9000";
```
Comment (tobegit3hub) [inline tensorflow_serving/example/inception_client.cc]: Make these parameters configurable

File: tensorflow_serving/example/inception_client.cc [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    }
+  } else {
+    image_path = INPUT_FILE_NAME;
+  }
+  return image_path;
+}
+
+
+int main(int argc, char** argv) {
+  // Expect only arg: --db_path=path/to/route_guide_db.json.
```
Comment (tobegit3hub) [inline tensorflow_serving/example/inception_client.cc]: I don't think we need this here

File: tensorflow_serving/example/inception_client.cc (hunk 1 above)
Comment (nfiedel) [inline tensorflow_serving/example/inception_client.cc]: Yes, please update SERVING_HOST to something like ADDRESS and make flag configurable.
Input file name should also be flag configurable.
And please s/SERVICE_NAME/MODEL_NAME, and make configurable. Can default to "inception".
Thanks!

File: tensorflow_serving/example/inception_client.cc [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    if (status.ok()) {
+      std::cout << "call predict ok" << std::endl;
+      std::cout << "outputs size is "<< response.outputs_size() << std::endl;
+      OutMap& map_outputs =  *response.mutable_outputs();
+      OutMap::iterator iter;
+      int output_index = 0;
+      for(iter = map_outputs.begin();iter != map_outputs.end(); ++iter){
+        std::cout << "output " << output_index << " name is "<< iter->first << std::endl;
+
+        std::cout << "output " << output_index << " type is "<<
```
Comment (nfiedel) [inline tensorflow_serving/example/inception_client.cc]: Looks like we can probably reuse helper methods from tensorflow/core/framework/tensor.cc, specifically 
Tensor::FromProto and Tensor::DebugString(). This should let you delete the code from here to approx line 120 of the current PR.

File: tensorflow_serving/example/inception_client.cc [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+      return "RPC failed";
+    }
+  }
+
+ private:
+  std::unique_ptr<PredictionService::Stub> stub_;
+};
+
+
+std::string GetImagePath(int argc, char** argv) {
```
Comment (nfiedel) [inline tensorflow_serving/example/inception_client.cc]: Can you please try tensorflow::Flag(...), which will allow you to delete this method?

**Comment by isiosia** [comment]
@nfiedel @tobegit3hub Thanks for your review, and I rewrote the code mentioned in the comments.The parameter processing and output are now using the methods provided by tensorflow. Would you please give another review?

**Comment by nfiedel** [comment]
Hi isiosia@,
  Thanks for making the changes. Looking good!

  @tensorflow-jenkins : test this please

**Comment by atisman89** [comment]
Is there a way to build the C++ gRPC client on Windows?

**Comment by syagev** [comment]
@atisman89 were you able to do this eventually?

**Comment by Opeiz** [comment]
Hi all,
I used these example with conda to run it but its says "grpc++/create_channel.h no such file or directory" 
anyone know how to fix it?

**Review by rasapala — COMMENTED** [review]
LGTM

**Pull request merged**